cmake/extensions.cmake

```diff
@@ -668,11 +684,22 @@ function(zephyr_check_compiler_flag lang option check)
   set(${check} ${inner_check} PARENT_SCOPE)
 
   # Populate the cache
-  file(
-    APPEND
-    ${ZEPHYR_TOOLCHAIN_CAPABILITY_CACHE}
-    "set(${key} ${inner_check})\n"
-    )
+  if(NOT (EXISTS ${key_path}))
+    file(
+      WRITE
+      ${key_path}
+      ${inner_check}
```

Comment:
Doesn't that just reduce the window of the race? I mean it's much better but you could still have two cmake instances trying to create the same key at the same time, no?

Reply:
I believe this is thread-safe.

AFAIK two threads writing the same byte to the same position of the same file is thread-safe.

But I could be wrong.

Reply:
OK so there is a hopefully harmless race but a race nevertheless. PR #15128 just submitted removes the "hopefully".